components/site/patterns/DeleteTable.js

```diff
@@ -5,59 +5,68 @@ import { Button } from '@twilio-paste/core/button';
 import { DeleteIcon } from '@twilio-paste/icons/cjs/DeleteIcon';
 import PropTypes from 'prop-types';
 
-export const DeleteTable = ({ data, handleDelete }) => {
-  if (data.length === 0) {
+export const DeleteTable = ({ columns, rows, emptyState, handleDelete }) => {
+  if (rows.length === 0) {
     return (
       <Text as="span" fontFamily="fontFamilyText">
-        Nothing to delete!
+        {emptyState}
       </Text>
     );
   }
 
-  const cols = Object.keys(data[0]);
+  rows.forEach((row) => {
+    if (row.length > columns.length)
+      throw new Error(
+        'If you are seeing this error, it is because you added more cells than there are columns. Go back to delete.js and add a new column heading. DO NOT try to fix this error in DeleteTable.js!',
+      );
+  });
 
   return (
     <Table>
       <THead>
         <Tr>
-          {cols.map((key) => {
-            if (key === 'id') return <></>;
-            return <Th key={cols.indexOf(key)}>{key}</Th>;
+          {columns.map((column) => {
+            return <Th key={columns.indexOf(column)}>{column}</Th>;
```

Comment:
`map` takes a second argument of `index` if you really want to use that as the key. Might save looking up the index of something in a collection.

`{columns.map((column, index) => {`

Reply:
@SiTaggart I didn't know that either! That does sound like a useful thing to use here. Great job figuring out a solution for the key prop, Olivia!

Reply:
implemented @SiTaggart 's suggestion in the resolve comment below, but this is good to remember in the future